q2_diversity/_filter.py

```diff
@@ -12,13 +12,16 @@
 
 def filter_distance_matrix(distance_matrix: skbio.DistanceMatrix,
                            metadata: qiime2.Metadata,
-                           where: str=None) -> skbio.DistanceMatrix:
+                           where: str=None,
+                           exclude_ids: bool=False) -> skbio.DistanceMatrix:
     ids_to_keep = metadata.ids(where=where)
     # NOTE: there is no guaranteed ordering to output distance matrix because
     # `ids_to_keep` is a set, and `DistanceMatrix.filter` uses its iteration
     # order.
     try:
+        if exclude_ids:
+            ids_to_keep = set(distance_matrix.ids) - set(ids_to_keep)
         return distance_matrix.filter(ids_to_keep, strict=False)
     except skbio.stats.distance.DissimilarityMatrixError:
-        raise ValueError(
-            "All samples were filtered out of the distance matrix.")
+        raise ValueError("All samples were filtered out of the "
+                         "distance matrix.")
```

Comment:
Also stylistic, but is this change necessary? Could it confuse someone looking at the git history in the future?